### PR TITLE
Add SatoshiProtocol Project

### PR DESCRIPTION
**Pull request by imfeng**
##### Name (to be shown on DefiLlama):
Satoshi Protocol

##### Twitter Link:
[https://twitter.com/SatoshiBEVM](https://twitter.com/SatoshiBEVM)

##### List of audit links if any:
[https://github.com/Satoshi-Protocol/satoshi-audit-report](https://github.com/Satoshi-Protocol/satoshi-audit-report)

##### Website Link:
https://satoshiprotocol.org/

##### Logo (High resolution, will be shown with rounded borders):
SatoshiProtocol

##### Current TVL:
57 K

##### Treasury Addresses (if the protocol has treasury)
N/A

##### Chain:
BEVM

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
N/A

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
N/A

##### Short Description (to be shown on DefiLlama):
Satoshi Protocol is the first CDP built on BTC Layer 2, it allows investors to deposit BTC and borrow stablecoin $SAT, Unleash trillion dollars of BTCFi.

##### Token address and ticker if any:
- **SAT** : [https://scan-mainnet.bevm.io/token/0xF2692468666E459D87052f68aE474E36C1a34fbB](https://scan-mainnet.bevm.io/token/0xF2692468666E459D87052f68aE474E36C1a34fbB)

##### Category (full list at https://defillama.com/categories) *Please choose only one:
CDP

##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:
[https://documents.bevm.io/build-decentralized-apps/integrations/oracle/dia](https://documents.bevm.io/build-decentralized-apps/integrations/oracle/dia
)

##### forkedFrom (Does your project originate from another project):
[Liquity Protocol](https://www.liquity.org/)

##### methodology (what is being counted as tvl, how is tvl being calculated):
assets in main contract.

##### Github org/user (Optional, if your code is open source, we can track activity):
[https://github.com/Satoshi-Protocol](https://github.com/Satoshi-Protocol)

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/satoshi-protocol.js exports TVL: 
        
 
 
    bevm                      57.22 k
    
    total                    57.22 k

**Comment by llamatester** [comment]
The adapter at projects/satoshi-protocol exports TVL: 
        
 
 
    bevm                      57.45 k
    
    total                    57.45 k

**Comment by g1nt0ki** [comment]
@imfeng thanks for the PR

File: test.js [hunk 1]
```diff
@@ -110,7 +110,8 @@ sdk.api.abi.call = async (...args) => {
   const chains = Object.keys(module).filter(item => typeof module[item] === 'object' && !Array.isArray(module[item]));
   checkExportKeys(module, passedFile, chains)
   const unixTimestamp = Math.round(Date.now() / 1000) - 60;
-  const { chainBlocks } = await getCurrentBlocks([]); // fetch only ethereum block for local test
+  // const { chainBlocks } = await getCurrentBlocks([]); // fetch only ethereum block for local test
+  const chainBlocks = {}
```
Comment (philipjames44) [inline test.js:114]: Why was this merged?

Comment (g1nt0ki) [reply test.js:114]: so, the test runs faster

**Pull request merged**